### DISCUSSION TITLE
Changing the ID in the URL

### DISCUSSION THREAD

**Discussion by onairmarc** [filamentphp/filament#12809] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.78

### How can we help you?

Is it possible to change the ID used in the URL within Filament? I'm using the `spatie/laravel-prefixed-ids` package and want the URL to reflect the prefixed ID and lookup using that instead of the ID column. So if I have an invoice with ID 100, instead of `/invoices/100/edit`, I would want the URL to be `/invoices/INV100/edit`.

I've reviewed [this documentation](https://filamentphp.com/docs/3.x/panels/resources/getting-started#generating-urls-to-resource-pages), which seems to be close to what I'm looking for, but it doesn't specifically mention if it's possible to change the column used. Any guidance here would be appreciated.

**Comment by akki-io** [comment, marked as answer]
You can use 

```
public function getRouteKeyName()
{
    return 'prefixed_id';
}
```

in your model. 

Filament will then use the `prefixed_id` column in the URL and it will auto-resolve.
